src/base/bittorrent/session.cpp

```diff
@@ -4064,6 +4069,35 @@ void Session::handleMetadataReceivedAlert(const lt::metadata_received_alert *p)
     }
 }
 
+void Session::handleTorrentPausedAlert(const libtorrent::torrent_paused_alert *p)
+{
+    const InfoHash hash {p->handle.info_hash()};
+
+    if (m_loadedMetadata.contains(hash)) {
+        // Adding preloaded torrent
+        m_loadedMetadata.remove(hash);
+        lt::torrent_handle handle = m_nativeSession->find_torrent(hash);
+        --m_extraLimit;
+
+        try {
```

Comment:
is it possible that it will throw exceptions?

Reply:
libtorrent docs describe the possibility of exceptions is somewhat unclear. In theory, this operation should not throw an exception if the handle is valid (and it should be valid in this place of the code by design). Apparently it was done to be safe. I can remove it the same way I previously did for (almost) all similar cases. And it doesn't seem to cause problems...

Reply:
>I can remove it the same way I previously did for (almost) all similar cases. And it doesn't seem to cause problems...

I'm fine with it.